core/src/main/scala/org/apache/spark/h2o/utils/H2OContextUtils.scala

```diff
@@ -51,6 +54,60 @@ private[spark] trait H2OContextUtils extends Logging {
     }
   }
 
+  private def isTcpPortAvailable(port: Int): Boolean = {
+    try {
+      val serverSocket = new ServerSocket()
+
+      serverSocket.setReuseAddress(false);
+      serverSocket.bind(new InetSocketAddress(InetAddress.getByName(SparkEnv.get.blockManager.blockManagerId.host), port), 1)
+      serverSocket.close()
+      true
+    } catch {
+      case _: Exception => false
+    } finally {
+    }
+  }
+
+  private def findNextFreeFlowPort(conf: H2OConf): Int = {
+    if (conf.clientWebPort == -1) {
+      var port = conf.clientBasePort
+      val offset = conf.internalPortOffset
+      while (!isTcpPortAvailable(port)) {
+        port = port + offset + 1
+      }
+      port
+    } else {
+      val port = conf.clientWebPort
+      if (!isTcpPortAvailable(port)) {
+        throw new RuntimeException(s"Explicitly specified client web port: $port is not available!")
+      } else {
+        port
+      }
+
+    }
+  }
+
+  protected def startFlowProxy(conf: H2OConf): URI = {
+    val port = findNextFreeFlowPort(conf)
+    val server = new Server(port)
+
+    val context = new ServletContextHandler(ServletContextHandler.SESSIONS);
+    context.setContextPath("/");
```

Comment:
Shouldn't we use context path from `H2OConf`?

Reply:
Not here, but should be specified in the `prefix` parameter